File: Map/scripts/extract_pdf_urls.py
import re
import sys

path = sys.argv[1]
data = open(path, "rb").read()
urls = re.findall(rb"https?://[^\s\)\>\]<\"']+", data)
for u in sorted(set(urls)):
    print(u.decode("latin-1", "ignore"))

# Also look for URI annotations
for m in re.finditer(rb"/URI\s*\(([^)]+)\)", data):
    print("URI:", m.group(1).decode("latin-1", "ignore"))

File: Map/scripts/extract-pdf-urls.mjs
import fs from "node:fs";

const pdfPath = process.argv[2];
const data = fs.readFileSync(pdfPath);
const text = data.toString("latin1");

const httpUrls = [...text.matchAll(/https?:\/\/[^\s\)\>\]<"']+/g)].map((m) => m[0]);
const uriAnnots = [...text.matchAll(/\/URI\s*\(([^)]+)\)/g)].map((m) => m[1]);

console.log("HTTP URLs:");
for (const u of [...new Set(httpUrls)]) console.log(u);

console.log("\nURI annotations:");
for (const u of [...new Set(uriAnnots)]) console.log(u);
